File: Data Structures & Algorithms/sliding-window-maximum/submission-4.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int n=nums.length;
        List<Integer> l=new ArrayList<>();
        for(int i=0;i<=n-k;i++){
            int mx=Integer.MIN_VALUE;
             for(int j=i;j<i+k;j++){
                mx=Math.max(mx,nums[j]);
             }
             l.add(mx);
        }
        int a[]=new int[l.size()];
        for(int i=0;i<l.size();i++){
            a[i]=l.get(i);
        }
        return a;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-4.py
class Solution:
    def search(self, nums: List[int], tar: int) -> int:
        l,h=0,len(nums)-1
        while l<=h:
            mid=(l+h)//2
            if nums[mid]==tar:
                return mid
            if nums[mid]>=nums[l]:
                if tar>nums[mid] or tar<nums[l]:
                    l=mid+1
                else:
                    h=mid-1
            else:
                if tar<nums[mid] or tar>nums[h]:
                    h=mid-1
                else:
                    l=mid+1
        return -1